release_tools/notes.py

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2019 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Santiago Dueñas <sduenas@bitergia.com>
+#
+
+"""
+Script to generate the release notes of a package.
+
+It will read the changelog entries stored under 'releases/unreleased'
+directory and will generate a file with the list of changes.
+The script needs the name of the package and the version to release.
+"""
+
+import datetime
+import os
+import sys
+import textwrap
+
+import click
+
+from release_tools.entry import (CategoryChange,
+                                 determine_changelog_entries_dirpath,
+                                 read_changelog_entries)
+from release_tools.repo import GitHandler
+
+
+def validate_argument(ctx, param, value):
+    """Check argument valid values."""
+
+    value = value.strip("\n\r ")
+
+    if not value:
+        raise click.BadParameter("cannot be empty")
+    return value
+
+
+@click.command()
+@click.option('--dry-run', is_flag=True,
+              help="Do not write release notes file. Print to the standard output instead.")
+@click.option('--overwrite', is_flag=True,
+              help="Force to replace an existing release notes entry.")
+@click.argument('name', callback=validate_argument)
+@click.argument('version', callback=validate_argument)
+def notes(name, version, dry_run, overwrite):
+    """Generate release notes.
+
+    When you run this script, it will generate the release notes of the
+    package tracked by the current Git repository.
+
+    You will need to provide the 'NAME' of the package and the 'VERSION'
+    of the new release. The script will generate a Markdown document
+    under the 'releases' directory. Take into account the argument `NAME`
+    is only used as the title of the document.
+
+    In the case a release notes file for the same version already exists,
+    an error will be raised. Use '--overwrite' to force to replace the
+    existing file. If you just want to see the final result of the notes
+    but not generate a new file, please activate '--dry-run' flag.
+
+    NAME: title of the package for the release notes.
```

Comment:
description of dry_run and overwrite are missing

Reply:
I did on purpose because the way click works. This text is shown as the output of `--help`, so the description of these fields is already included.